lib/graphql/query/partial.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+module GraphQL
+  class Query
+    # This class is _like_ a {GraphQL::Query}, except
+    # @see Query#run_partials
+    class Partial
+      def initialize(path:, object:, query:)
+        @path = path
+        @object = object
+        @query = query
+        @context = GraphQL::Query::Context.new(query: self, schema: @query.schema, values: @query.context.to_h)
+        @multiplex = nil
+        @result_values = nil
+        @result = nil
+      end
+
+      attr_reader :context
+
+      attr_accessor :multiplex, :result_values
+
+      def result
+        @result ||= GraphQL::Query::Result.new(query: self, values: result_values)
+      end
+
+      def valid?
+        true
+      end
+
+      def analyzers
+        EmptyObjects::EMPTY_ARRAY
+      end
+
+      def current_trace
+        @query.current_trace
+      end
+
+      def analysis_errors=(_errs)
+      end
```

Comment:
TODO raise an error if this is actually assigned; or find some way to skip analysis checks altogether; it's not supported.